linera-core/src/client/mod.rs

```diff
@@ -1981,11 +1990,19 @@ impl<Env: Environment> ChainClient<Env> {
         let creating_proposal_start = Instant::now();
         let info = self.chain_info().await?;
         let timestamp = self.next_timestamp(incoming_bundles, info.timestamp);
+        let transactions = incoming_bundles
+            .iter()
+            .map(|bundle| Transaction::ReceiveMessages(bundle.clone()))
+            .chain(
+                operations
+                    .iter()
+                    .map(|operation| Transaction::ExecuteOperation(operation.clone())),
+            )
```

Comment:
Why not do that with a `From<_>` impl and do sth like `incoming_bundles.iter().map(Into::into).chain(...)`.